NuGet.config

```diff
@@ -12,13 +12,22 @@
     <!--  Begin: Package sources from dotnet-runtime -->
     <!--  End: Package sources from dotnet-runtime -->
     <!--End: Package sources managed by Dependency Flow automation. Do not edit the sources above.-->
+    <!-- TEST_RESTORE_SOURCES_INSERTION_LINE -->
     <add key="dotnet-public" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-public/nuget/v3/index.json" />
     <add key="dotnet-eng" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json" />
     <add key="dotnet8" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet8/nuget/v3/index.json" />
     <add key="dotnet-libraries" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-libraries/nuget/v3/index.json" />
     <add key="orleans-nightly" value="https://orleans.pkgs.visualstudio.com/orleans-public/_packaging/orleans-nightly/nuget/v3/index.json" />
+    <add key="dotnet9-transport" value="https://dnceng.pkgs.visualstudio.com/public/_packaging/dotnet9-transport/nuget/v3/index.json" />
   </packageSources>
   <packageSourceMapping>
+    <!-- used for installing test workloads for resolving aspire workload manifest -->
+    <packageSource key="nuget-local">
```

Comment:
https://github.com/dotnet/runtime/blob/f135699310bccfa83b3b441d2f4f1571fe8df480/src/tasks/WorkloadBuildTasks/InstallWorkloadFromArtifacts.cs#L262

I agree that it does seem odd that you can't directly see what this links up to in the repo. I can have the package source name be passed to the task, which would be in aspire repo.

Reply:
Does this throw a warning at all on the initial restore before the task has a chance to replace the source? Did we also consider manually passing in `-s <path>` to a manual invocation of restore instead of having this in the config?

Reply:
No warnings. Wouldn't `-s path` only add a package source but not the mapping, and because of that restoring from the source would fail. That is what I hit when I removed this mapping.

Reply:
Once https://github.com/dotnet/runtime/pull/99277 is merged this entry can be removed, and it would be added at build time.